### ISSUE TITLE
Is it possible to filter certain module inside a tracer.wrap or tracer.trace ? 

### ISSUE THREAD

**Issue by HassankSalim** [DataDog/dd-trace-py#3613]
Is it possible to filter or downsample only certain modules (like Django or Postgres) inside a tracer.wrap or tracer.trace without effected the global configuration? 

Use case or Experience: While wrapping function which does huge background processing, the number of SQL queries can be high, leading to a trace having 40k spans, then when dd-trace-py sends the trace it fails at datadog agent level and the entire trace info is lost.  I would like to only trace custom functions that I have traced through `tracer.trace`

I have confirmed that it's the high span count which causes the failure by disabling patching of Django and Postgres using the env variables, which worked 
but this is not a  practical solution since the env variables disable the traces for all the APIs and functions (its a global configure)

Is there a way around this? 

### What is the result that you expected?

Ideally, the solution would be to disable some modules or downsample some modules at a particular trace level. For example, a configuration could be passed to `tracer.wrap` or `tracer.trace`

**Comment by HassankSalim** [comment]
One possible solution is 

```
from ddtrace import tracer
from ddtrace import patch_all
from ddtrace.filters import  TraceFilter
import requests

patch_all()

tracer.configure(settings={
       'FILTERS': [
                RequestFilter()
       ],
})

class RequestFilter(TraceFilter):
    def process_trace(self, trace):
        # type: (List[Span]) -> Optional[List[Span]]
        filter_out_traces = []
        ignore_request_trace = False
        for span in trace:
            if span.get_tag('ignore_request'):
                ignore_request_trace = True
                break

        if ignore_request_trace:
            for span in trace:
                if span.span_type != 'http' and span.name != 'requests.request':
                    filter_out_traces.append(span)
        return filter_out_traces if ignore_request_trace else trace


@tracer.wrap()
def get_request_without_trace():
    span = tracer.current_span()
    span.set_tag('ignore_request', True)
    resp = requests.get('https://www.example.com')
    return resp.status_code


@tracer.wrap()
def get_request():
    resp = requests.get('https://www.example.com')
    return resp.status_code
```


Is there a better way ?

**Comment by Kyle-Verhoog** [comment]
hi @HassankSalim, sorry for the very late reply! The snippet you posted should work. We have recently added support for single span sampling which we're about to GA in ddtrace 1.7 (but is available in ddtrace 1.6). It requires Datadog agent 7.40 as well.

The setting to use is `DD_SPAN_SAMPLING_RULES`

```
DD_SPAN_SAMPLING_RULES
    Default: []
    A JSON array of objects. Rules are applied in configured order to determine the span’s sample rate. The sample_rate value must be between 0.0 and 1.0 (inclusive). For more information, see [Ingestion Mechanisms](https://docs-staging.datadoghq.com/antoine.dussault/single_spans/tracing/trace_pipeline/ingestion_mechanisms/).
    Example:

        Set the span sample rate to 50% for the service my-service and operation name http.request, up to 50 traces per second: '[{"service": "my-service", "name": "http.request", "sample_rate":0.5, "max_per_second": 50}]'
```

([preview doc link](https://docs-staging.datadoghq.com/antoine.dussault/single_spans/tracing/trace_collection/library_config/python/))


Going to mark the issue as resolved but please yell if there's anything further 🙂
